load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,122 +15,224 @@
+  @Parameter(
+      names = {"--max_concurrent_requests"},
+      description = "The maximum number of concurrent requests each client will allow."
+  )
+  private int maxConcurrentRequests = 20;
+  @Parameter(
+      names = {"--burn_in_duration_seconds"},
+      description = "The duration, in seconds, to run without recording statistics in order to allow tuning."
+  )
+  private int burnInDurationSeconds = 20;
+  @Parameter(
+      names = {"--number_of_messages"},
+      description = "The number of messages to publish in the test. Enabling this will override " +
+          "--loadtest_length_seconds. Enabling this flag will also enable the check for message loss. If set less " +
+          "than 1, this flag is ignored."
+  )
+  private int numberOfMessages = 0;
 
   public static void main(String[] args) {
     // Turns off all java.util.logging.
     LogManager.getLogManager().reset();
+    Driver driver = new Driver();
+    JCommander jCommander = new JCommander(driver, args);
+    if (driver.help) {
+      jCommander.usage();
+      return;
+    }
+    driver.run();
+  }
+
+  private void run() {
     try {
-      // Parse command line arguments.
-      BaseArguments baseArgs = new BaseArguments();
-      CPSArguments cpsArgs = new CPSArguments();
-      KafkaArguments kafkaArgs = new KafkaArguments();
-      CompareArguments dataComparisonArgs = new CompareArguments();
-      JCommander jCommander = new JCommander(baseArgs);
-      jCommander.addCommand(CPSArguments.COMMAND, cpsArgs);
-      jCommander.addCommand(KafkaArguments.COMMAND, kafkaArgs);
-      jCommander.addCommand(CompareArguments.COMMAND, dataComparisonArgs);
-      jCommander.parse(args);
-      if (jCommander.getParsedCommand() == null) {
-        if (baseArgs.isHelp()) {
-          jCommander.usage();
-        }
-        return;
-      }
-      if (jCommander.getParsedCommand().equals(CompareArguments.COMMAND)) {
-        // Compares data dumps for correctness.
-        Comparison c = new Comparison(dataComparisonArgs.getFile1(), dataComparisonArgs.getFile2());
-        c.compare();
-        return;
+      Map<String, Map<ClientParams, Integer>> clientTypes = ImmutableMap.of(
+          "us-central1-a", new HashMap<>());
+      Preconditions.checkArgument(
+          cpsPublisherCount > 0 ||
+              cpsSubscriberCount > 0 ||
+              kafkaPublisherCount > 0 ||
+              kafkaSubscriberCount > 0
+      );
+      Preconditions.checkArgument(
+          broker != null || (kafkaPublisherCount == 0 && kafkaSubscriberCount == 0));
+      for (int i = 0; i < subscriberFanout; ++i) {
+        clientTypes.get("us-central1-a").put(new ClientParams(ClientType.CPS_GCLOUD_PUBLISHER, null),
+            cpsPublisherCount / subscriberFanout);
+        clientTypes.get("us-central1-a").put(new ClientParams(ClientType.CPS_GCLOUD_SUBSCRIBER,
+                "gcloud-subscription" + i),
+            cpsSubscriberCount / subscriberFanout);
```

Comment:
Correct. That is built in, as users cannot set topics. This seems like something I should add to the Readme?